[[ Bug 20061 ]] Save m_rep in unbound widget

montegoulding opened the pull request:
When a widget bind fails we retain the loaded rep. This patch
ensures that that the retained rep is saved instead of a null
rep.

montegoulding committed:
[[ Bug 20061 ]] Save m_rep in unbound widget

When a widget bind fails we retain the loaded rep. This patch
ensures that that the retained rep is saved instead of a null
rep.

montegoulding commented:
This may be possible to test with a bit of jiggery pokery loading, saving, deleting, unloading, opening, saving, deleting, loading then checking data...

runrevmark commented:
I think the steps would be:
- make a widget with a foo property
- load widget module
- create widget on stack
- save stack
- delete stack
- unload widget module
- load stack
- save stack
- delete stack
- load widget module
- load stack
- check foo property value

runrevmark commented:
I'd forgotten that we've got 'export widget to array arrayVar' - so we can use that to test even more easily :)

montegoulding commented:
@runrevmark so using import/export widget I guess the test can come down to import non-existent widget with some state then export and check state... I wonder if that's a complete test as export doesn't save does it

montegoulding commented:
Looking at export I think it should also use `m_rep` if bind failed. At present it doesn't.

montegoulding committed:
[[ Bug ]] Load aux extensions relative to caller

This patch fixes a bug where auxillary extensions were being loaded
relative to `the filename of this stack`. This is unhelpful if the
test has opened other stacks and may not now be the default stack.

runrevmark commented:
@livecode-vulcan review ok e68f494

montegoulding commented:
Hmm... vulcan appears to be ignoring this

livecodeali commented:
@livecode-vulcan review ok 42acb91

livecodeali commented:
I think there's something up with the timestamps

livecode-vulcan commented:
💙 review by @livecodeali ok 42acb9113790e5018193ce128251acf4c28f8b45

livecode-vulcan commented:
😎 test success 42acb9113790e5018193ce128251acf4c28f8b45

* try-community-armv6-android-api8: [success](https://vulcan.livecode.com/builders/try-community-armv6-android-api8/builds/522)
* try-community-armv6-android-api9: [success](https://vulcan.livecode.com/builders/try-community-armv6-android-api9/builds/537)
* try-community-js-emscripten-sdk1.35: [success](https://vulcan.livecode.com/builders/try-community-js-emscripten-sdk1.35/builds/511)
* try-community-universal-ios-iphoneos10.3: [success](https://vulcan.livecode.com/builders/try-community-universal-ios-iphoneos10.3/builds/454)
* try-community-universal-ios-iphonesimulator10.3: [success](https://vulcan.livecode.com/builders/try-community-universal-ios-iphonesimulator10.3/builds/457)
* try-community-universal-mac-macosx10.6: [success](https://vulcan.livecode.com/builders/try-community-universal-mac-macosx10.6/builds/517)
* try-community-universal-mac-macosx10.9: [success](https://vulcan.livecode.com/builders/try-community-universal-mac-macosx10.9/builds/525)
* try-community-x86-linux-debian7: [success](https://vulcan.livecode.com/builders/try-community-x86-linux-debian7/builds/520)
* try-community-x86-linux-debian8: [success](https://vulcan.livecode.com/builders/try-community-x86-linux-debian8/builds/537)
* try-community-x86_64-linux-debian7: [success](https://vulcan.livecode.com/builders/try-community-x86_64-linux-debian7/builds/520)
* try-community-x86_64-linux-debian8: [success](https://vulcan.livecode.com/builders/try-community-x86_64-linux-debian8/builds/530)
* try-community-x86-win32: [success](https://vulcan.livecode.com/builders/try-community-x86-win32/builds/552)
* try-community-x86_64-win32: [success](https://vulcan.livecode.com/builders/try-community-x86_64-win32/builds/515)
